protocol/auditor.go

```diff
@@ -0,0 +1,55 @@
+// This module implements a generic CONIKS auditor, i.e. the
+// functionality that clients and auditors need to verify
+// a server's STR history.
```

Comment:
I see. I thought we were going to merge this PR first. It's certainly closer to being done. This still duplicates code from #170, though, so in either case, I think the generic auditor code should be removed from this PR since it will be added as part of #170.